component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java

```diff
@@ -0,0 +1,290 @@
+                              boolean excludeEmptyNavigationSites,
+                              @Parameter(description = "Used to retrieve the site label and description in the requested language")
+                              @QueryParam("lang")
+                              String lang) {
+    try {
+      PortalConfig site = layoutService.getPortalConfig(Long.parseLong(siteId));
+      return Response.ok(EntityBuilder.buildSiteEntity(site,
+                                                       request,
+                                                       expandNavigations,
+                                                       visibilityNames,
+                                                       excludeEmptyNavigationSites,
+                                                       temporalCheck,
+                                                       excludeGroupNodesWithoutPageChildNodes,
+                                                       getLocale(lang)))
+                     .build();
+    } catch (Exception e) {
+      LOG.warn("Error while retrieving site", e);
+      return Response.serverError().build();
+    }
+  }
+
+  @GET
+  @Path("{siteName}/banner")
+  @RolesAllowed("users")
+  @Produces("image/png")
+  @Operation(summary = "Gets a site banner", method = "GET")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "500", description = "Internal server error"),
+      @ApiResponse(responseCode = "403", description = "Forbidden request"),
+      @ApiResponse(responseCode = "404", description = "Resource not found") })
+  public Response getSiteBanner(@Context
+  Request request,
+                                @Parameter(description = "site name", required = true)
+                                @PathParam("siteName")
+                                String siteName,
+                                @Parameter(description = "A mandatory valid token that is used to authorize anonymous request", required = true)
+                                @QueryParam("r")
+                                String token,
+                                @Parameter(description = "A mandatory valid token that is used to determinate if to use default banner", required = true)
+                                @QueryParam("isDefault")
+                                boolean isDefault) {
+    try {
+      if (!isDefault && RestUtils.isAnonymous()
+          && !LinkProvider.isSiteBannerTokenValid(token, siteName, LinkProvider.ATTACHMENT_BANNER_TYPE)) {
+        LOG.warn("An anonymous user attempts to access banner of site {} without a valid access token", siteName);
+        return Response.status(Response.Status.FORBIDDEN).build();
+      }
+      InputStream stream = isDefault ? getDefaultBannerInputStream(siteName) : layoutService.getSiteBannerStream(siteName);
+      return Response.ok(stream, "image/png").build();
```

Comment:
Same comment as previous for `ETag` usage.

Reply:
updated